src/utils.rs

```diff
@@ -0,0 +1,31 @@
+use anyhow::Context;
+use reqwest::blocking::Response;
+use serde::de::DeserializeOwned;
+
+pub trait ResponseExt {
+    fn custom_error_for_status(self) -> anyhow::Result<Response>;
+    fn json_annotated<T: DeserializeOwned>(self) -> anyhow::Result<T>;
+}
+
+impl ResponseExt for Response {
+    fn custom_error_for_status(self) -> anyhow::Result<Response> {
+        match self.error_for_status_ref() {
+            Ok(_) => Ok(self),
+            Err(err) => {
+                let body = self.text()?;
+                Err(err).context(format!("Body: {:?}", body))
```

Comment:
This would put the "Body" error as the top error type. Maybe the following:

```suggestion
                anyhow!("Body for error: {body:?}").context(err)
```

Reply:
Btw, the wording is weird if I switch the contexts:
```
[2023-12-08T17:22:00Z ERROR sync_team] failed: HTTP status client error (400 Bad Request) for url (https://httpstat.us/400)
[2023-12-08T17:22:00Z ERROR sync_team] caused by: Body for error: "400 Bad Request"
```
it's not "caused by" the custom body content.